examples/resources/kentik-synthetic_test-bulk-create/resource.tf

```
@@ -0,0 +1,92 @@
+// Bulk synthetic tests create use cases:
+// 1. Customer wants to automatically create synthetic tests from a set of agents
+//    to addresses of interfaces classified as “external”.
+// 2. Customer wants to automatically create synthetic tests to his applications hosted in AWS
+//    from all of his private agents.
+// Use case #2 below.
+
+data "kentik-synthetics_agents" "agents" {}
+
+locals {
+  agents = [
+    for agent in data.kentik-synthetics_agents.agents:
+      agent
+  ]
+  agents_data = local.agents[2]
+  private_agents_ids = compact([for agent in local.agents_data : agent.type == "private" ? agent.id : ""])
+}
+
+resource "kentik-synthetics_test" "private-agents-test" {
+  for_each = toset(local.private_agents_ids)
+
+  name      = "private-agents-test"
+  type      = "hostname"
+  device_id = "75702"
+  status    = "TEST_STATUS_ACTIVE"
+  settings {
+    hostname {
+      target = "www.example.com"
+    }
+    agent_ids = [each.value]
```

```suggestion
    agent_ids = local.private_agents_ids
```
Is it possible to assign agent IDs directly in this line without using `for_each`? It would be simpler.

I looked at docs of `for_each` and I think it is meant to be used to define multiple resource instances based on input collection. And we need only one instance of the `private-agents-test`.

---

Yes, that is possible, i just assumed that it should create a`private-agents-test` per agent. But now as I read the name itself - `private-agents-test` - it seems it should be done the way you propose it :P

---

Addressed with 01e0e6f6ceae4e23e62d3c30faa17c4adf894c68